### PR TITLE
`azurerm_container_registry`: add support of `export_policy_enabled`

### LINKED ISSUE #14514: `azurerm_container_registry` Support for exportPolicy
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I am currently hitting an Azure policy `Container registries should have exports disabled`. This property is only supported since api `2021-06-01-preview` and I noticed the provider is currently using `2020-11-01-preview`. Is there a default value being set when the property isn't supported in the API? Because my deployments are failling the Azure policy requirement even after manually setting `properties.policies.exportPolicy.status=disabled` in the existing resource.

In any case, my feature request is thus the ability to set exportPolicy parameter which would require at least api version `2021-06-01-preview`. 

Additional question, are the api versions just being updated "when needed"?

### New or Affected Resource(s)

* azurerm_container_registry

### Potential Terraform Configuration

```hcl
resource "azurerm_container_registry" "acr" {
  name                = "containerRegistry1"
  resource_group_name = azurerm_resource_group.rg.name
  location            = azurerm_resource_group.rg.location
  sku                 = "Premium"
  
  ...

  export_policy    { 
    enabled = false
  }
}
```

### References

* [Disable export of artifacts from an Azure container registry](https://docs.microsoft.com/en-us/azure/container-registry/data-loss-prevention)

### PR DESCRIPTION
**Pull request by magodo**
Fixes #14514

## Test

```bash
💢  TF_ACC=1 go test -v -timeout=20h ./internal/services/containers -run='TestAccContainerRegistry_policies|TestAccContainerRegistry_basic_basic'
=== RUN   TestAccContainerRegistry_basic_basic
=== PAUSE TestAccContainerRegistry_basic_basic
=== RUN   TestAccContainerRegistry_basic_basic2Premium2basic
=== PAUSE TestAccContainerRegistry_basic_basic2Premium2basic
=== RUN   TestAccContainerRegistry_policies
=== PAUSE TestAccContainerRegistry_policies
=== CONT  TestAccContainerRegistry_basic_basic
=== CONT  TestAccContainerRegistry_policies
=== CONT  TestAccContainerRegistry_basic_basic2Premium2basic
--- PASS: TestAccContainerRegistry_basic_basic (143.36s)
--- PASS: TestAccContainerRegistry_basic_basic2Premium2basic (206.96s)
--- PASS: TestAccContainerRegistry_policies (217.27s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/containers    217.302s
```

### CODE REVIEW

File: internal/services/containers/container_registry_resource.go [hunk 1]
```diff
@@ -385,6 +391,16 @@ func resourceContainerRegistry() *pluginsdk.Resource {
 				return fmt.Errorf("ACR trust policy can only be applied when using the Premium Sku. If you are downgrading from a Premium SKU please set trust_policy {}")
 			}
 
+			exportPolicyEnabled := d.Get("export_policy_enabled").(bool)
+			if !exportPolicyEnabled {
+				if !strings.EqualFold(sku, string(containerregistry.SkuNamePremium)) {
+					return fmt.Errorf("ACR export policy can only be disabled when using the Premium Sku. If you are downgrading from a Premium SKU please unset `export_policy_enabled` or set `export_policy_enabled = true`")
+				}
```
Comment (tombuildsstuff) [inline internal/services/containers/container_registry_resource.go:398]: this seems reasonable for the user to do in a single apply, but we may need to call the Update API twice to make that change?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
One comment around the update but this otherwise LGTM 👍

File: internal/services/containers/container_registry_resource.go (hunk 1 above)
Comment (magodo) [reply internal/services/containers/container_registry_resource.go:398]: @tombuildsstuff Is that covered by the test case [TestAccContainerRegistry_policies](https://github.com/hashicorp/terraform-provider-azurerm/blob/1f38fda7743d6319f1f231666468c72399d400f0/internal/services/containers/container_registry_resource_test.go#L450), where one update is enough?

**Comment by magodo** [comment]
@tombuildsstuff I've resolved the merge conflicts, please take another look!

**Review by katbyte — APPROVED** [review]
Thanks @magodo - LGTM 🖍️

**Pull request merged**